src/library_glemu.js

```diff
@@ -3125,6 +3125,9 @@ var LibraryGLEmulation = {
   glClientActiveTexture__sig: 'vi',
   glClientActiveTexture: function(texture) {
     GLImmediate.clientActiveTexture = texture - 0x84C0; // GL_TEXTURE0
+    if (GLImmediate.currentMatrix >= 2) { // Update the matrix stack when matrix mode is GL_TEXTURE.
+      GLImmediate.currentMatrix = 2 + GLImmediate.clientActiveTexture;
```

Comment:
I'm not a GL expert, but reading this I'm confused where the constant `2` comes from?

Reply:
I got it from the glMatrixMode emulation code. Matrix 0 is MODELVIEW, matrix 1 is PROJECTION, and >= 2 are the texture matrices.

Reply:
I see, thanks!